hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/util/ConcurrentMemoizingSupplier.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.impl.util;
+
+import java.util.function.Supplier;
+
+/**
+ * Wraps a {@code Supplier} and returns a thread-safe memoizing supplier which calls
```

Comment:
Can you limit the text line length to 72 chars here? (I mean the net length of the text line without `* `).